### PR TITLE
Timeseries: Utilize min/max on stacking percentage

### PR DESCRIPTION
**Pull request by gelicia**
**What is this feature?**

In timeseries, if you had a stacking percentage and then attempted to use min/max to define a range, the graph would disappear.

**Why do we need this feature?**

So the graph does not disappear when you attempt to define a min/max in this scenario.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/support-escalations/issues/13119

**Special notes for your reviewer:**

1. Test datasource, one series
2. Timeseries panel, set Graph Styles > Stack series to 100%
3. Set a min and max.

Note, to see it working you need to set your min and max values to the percentage - for example 0.2 to 0.4 . This can have a better user experience but I recommend we tackle that outside of just getting the panel to display.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: public/app/core/components/TimeSeries/utils.ts [hunk 1]
```diff
@@ -250,8 +250,8 @@ export const preparePlotConfigBuilder: UPlotConfigPrepFn = ({
           range:
             customConfig.stacking?.mode === StackingMode.Percent
               ? (u: uPlot, dataMin: number, dataMax: number) => {
-                  dataMin = dataMin < 0 ? -1 : 0;
-                  dataMax = dataMax > 0 ? 1 : 0;
+                  dataMin = field.config.min != null ? field.config.min : dataMin < 0 ? -1 : 0;
```
Comment (leeoniya) [inline public/app/core/components/TimeSeries/utils.ts:253]: maybe `dataMin = (field.config.min ?? dataMin) < 0 ? -1 : 0;`

**Comment by leeoniya** [comment]
reviewed breifly from phone, but let me check this better tomorrow. this range fn is pretty tempermental

**Comment by leeoniya** [comment]
@gelicia just pushed a different approach.

instead of fixing the custom range function for the percent stacked case in timeseries. i moved the stacked logic to the shared y scale function which already handles hard/soft min/max limits and a bunch of other scaling settings. this way it can be used more generally.

let me know what you think, or questions :)

**Review by leeoniya — APPROVED** [review]

**Comment by gelicia** [comment]
Wanted to double check how this might affect other visualizations that have percent stacking - the barchart is all good but I am seeing weird results on the histogram. If we would rather push this out and fix the histogram later (it is weird the same way in play and as far as I know, no one has reported a problem with it) I would be okay with that.

**Pull request merged**